Tests/AmplitudeTests.m

```diff
@@ -96,97 +108,90 @@ - (void)testInitializeLoadNilUserIdFromEventData {
-    XCTAssertEqual([newDBHelper2 getIdentifyCount], 0);
-    XCTAssertEqual([oldDbHelper getIdentifyCount], 2);
-
+    XCTAssertEqualObjects([[NSUserDefaults standardUserDefaults] objectForKey:[Amplitude getDataStorageKey:@"device_id" instanceName:newInstance1]], @"fakeDeviceId");
+    XCTAssertNotEqualObjects([[NSUserDefaults standardUserDefaults] objectForKey:[Amplitude getDataStorageKey:@"device_id" instanceName:newInstance2]], @"fakeDeviceId");
+    XCTAssertEqualObjects([[NSUserDefaults standardUserDefaults] objectForKey:[Amplitude getDataStorageKey:@"device_id" instanceName:kAMPDefaultInstance]], @"oldDeviceId");
+    [AMPStorage storeIdentify:@"{\"event_type\":\"$identify\"}" instanceName:newInstance1];
+    XCTAssertEqual([[self.amplitude getAllIdentifyWithInstanceName:newInstance1] count], 1);
+    XCTAssertEqual([[self.amplitude getAllIdentifyWithInstanceName:newInstance2] count], 0);
+    XCTAssertEqual([[self.amplitude getAllIdentify] count], 2);
+    
     [[Amplitude instanceWithName:newInstance2] setDeviceId:@"brandNewDeviceId"];
     [[Amplitude instanceWithName:newInstance2] flushQueue];
-    XCTAssertEqualObjects([newDBHelper1 getValue:@"device_id"], @"fakeDeviceId");
-    XCTAssertEqualObjects([newDBHelper2 getValue:@"device_id"], @"brandNewDeviceId");
-    XCTAssertEqualObjects([oldDbHelper getValue:@"device_id"], @"oldDeviceId");
-    [newDBHelper2 addEvent:@"{\"event_type\":\"testEvent2\"}"];
-    [newDBHelper2 addEvent:@"{\"event_type\":\"testEvent3\"}"];
-    XCTAssertEqual([newDBHelper1 getEventCount], 0);
-    XCTAssertEqual([newDBHelper2 getEventCount], 2);
-    XCTAssertEqual([oldDbHelper getEventCount], 1);
-
-    [newDBHelper1 deleteDB];
-    [newDBHelper2 deleteDB];
+    XCTAssertEqualObjects([[NSUserDefaults standardUserDefaults] objectForKey:[Amplitude getDataStorageKey:@"device_id" instanceName:newInstance1]], @"fakeDeviceId");
+    XCTAssertEqualObjects([[NSUserDefaults standardUserDefaults] objectForKey:[Amplitude getDataStorageKey:@"device_id" instanceName:newInstance2]], @"brandNewDeviceId");
+    XCTAssertEqualObjects([[NSUserDefaults standardUserDefaults] objectForKey:[Amplitude getDataStorageKey:@"device_id" instanceName:kAMPDefaultInstance]], @"oldDeviceId");
+    
+    [AMPStorage storeEvent:@"{\"event_type\":\"testEvent2\"}" instanceName:newInstance2];
+    [AMPStorage storeEvent:@"{\"event_type\":\"testEvent2\"}" instanceName:newInstance2];
+    XCTAssertEqual([[self.amplitude getAllIdentifyWithInstanceName:newInstance1] count], 1);
+    XCTAssertEqual([[self.amplitude getAllIdentifyWithInstanceName:newInstance2] count], 0);
+    XCTAssertEqual([[self.amplitude getAllEventsWithInstanceName:newInstance2] count], 2);
+    XCTAssertEqual([[self.amplitude getAllEventsWithInstanceName:kAMPDefaultInstance] count], 1);
 }
 
 - (void)testInitializeLoadUserIdFromEventData {
-    NSString *instanceName = @"testInitialize";
+    NSString *instanceName = @"testinitialize";
```

Comment:
For the all lowercase here, I see we are using it with Amplitude initialization, would it matters if we initialize with not pure lowercase string? I mean for customers who use this they may not just use all lowercases.

Reply:
we do allows all cases for instance name, however, the _instance name will be changed to lower case after the initialization. So when we later call getDataStorageKey later it’s always be the lower case.
So here I just directly force the instance name to be lower case.

Reply:
Oh, I see, so you don't need to pass lowercase when asserting.